src/dra/5298C.c

```diff
@@ -47,14 +47,69 @@ bool IsAlucart(void) {
 
 INCLUDE_ASM("asm/us/dra/nonmatchings/5298C", func_800F4994);
 
-INCLUDE_ASM("asm/us/dra/nonmatchings/5298C", func_800F4D38);
+s32 CalcAttack(s32 equipId, s32 otherEquipId) {
+    s32 i;
+    u16 equipmentAttackBonus;
+    s16 totalAttack;
+    s16 strengthStat;
+
+    if (D_800A4B04[equipId].damageScale == 6 ||
+        D_800A4B04[equipId].damageScale == 10 ||
+        (D_800A4B04[equipId].damageScale == 9 &&
+         D_800A4B04[equipId].attack == 1)) {
+        return 0;
+    }
+    if (equipId == 0x10) {
+        return 0;
+    }
+    equipmentAttackBonus = 0;
+    for (i = 0; i < 5; i++) {
+        equipmentAttackBonus +=
+            (u16)D_800A7718[g_Status.equipment[2 + i]].attBonus;
+    }
+
+    totalAttack = D_800A4B04[equipId].attack;
+    strengthStat = g_Status.statsTotal[0];
+
+    if (strengthStat >= totalAttack) {
+        totalAttack += strengthStat;
+    } else {
+        totalAttack += strengthStat / 2;
+    }
+
+    totalAttack += equipmentAttackBonus;
+
+    if (equipId == 0x7D) { // Badelaire sword
+        totalAttack += g_Status.timerHours;
+    }
+    if (equipId == 0x8D) { // Muramasa sword
+        totalAttack += SquareRoot0(g_Status.D_80097C40);
```

Comment:
D_80097C40 -> numMuramasaBloodHeals or similar. I checked by changing 97c00 in main ram to 0x8d, re-equipping in the menus to load the sword, then killing an enemy to heal. D_80097C40 goes up by one each time.

Reply:
Agree, though I'd like to focus on getting this PR in first, then allow a check in names-discussion to make sure others agree. I'm also looking into anywhere else in the code that references this address, would be nice to have all functions that use it dealt with before names start changing. For now I'd like to leave it as the D_ value.